### DISCUSSION TITLE
How to change color theme?

### DISCUSSION THREAD

**Discussion by ags1234** [FarGroup/FarManager#812] [Q&A]
Using Far decompressed in a directory from Far30b6286.x86.20240315.7z archive.
In Addons\Colors\Interface\ noticed some predefined themes, but it is not obvious to me how to apply them.
Went to Options - Colors and changed a few things there that reflected in the interface, but I don't know how to save all my modifications to a separate theme.

**Comment by alabuzhev** [comment]
> it is not obvious to me how to apply them

See Addons\README.TXT

> how to save all my modifications to a separate theme

No predefined action for this, but you can export the configuration (see far /?) and copy the corresponding node into a new file.

**Comment by ags1234** [comment]
Ah, thanks @alabuzhev !
So assuming you opened Command Prompt and are in the current directory where far.exe is:

`Far.exe /import .\Addons\Colors\Interface\dn_like.farconfig`

to import the configuration from the desired file, `dn_like.farconfig` in this case. Then, to start the program with a configuration that includes what we imported above:

`Far.exe`

Just trying out Far and it seems nice, though a little awkward having just a text interface.
A little annoying that it does not fit as much content as a graphical orthodox file manager. So I have to make the window bigger to see all the long names.
And it does not seem too advanced with color theme management.

**Reply by alabuzhev** [reply]
> I have to make the window bigger to see all the long names

You can make the window bigger, or the font smaller, or the panel or its columns wider, or scroll the long names manually, there are quite a few options.

> And it does not seem too advanced with color theme management

In theory we could introduce a proper UI for this, with a list of available themes, preview, export etc.
I even have a draft branch somewhere.
However, usually you only try other themes once, realize that you don't like them anyway, come up with your own and stick to it till the end of time, so there's not much motivation to make this too advanced.

**Reply by GetNight** [reply]
> So I have to make the window bigger to see all the long names.

In order to use it comfortably, the shortcut to the programme needs a little tweaking. But my system does not allow me to change the shortcut in the start menu. So I choose a location (right mouse button, properties), open C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Far Manager 3 (x64) - with the Shift key pressed I drag the shortcut to the desktop, change it there (with confirmation from the administrator) and bring it back. Then I link it to the start menu and the desktop.
What I change in the settings of the shortcut for comfortable use: 1. shortcut/window - I change from ‘expanded to full screen’ to ‘normal window size’; 2. shortcut/additional: ‘run as administrator’; 3. font = 14 ‘Consolas’; 4. location/size of the window 200/60. You get a very convenient interface that does not overlap other windows. Besides, it can always be ‘pushed’ aside. I also put ‘quick call’ = Ctrl+Alt+F. And at any second I can call FAR by pressing this combination.

**Comment by ags1234** [comment]
> However, usually you only try other themes once, realize that you don't like them anyway, come up with your own and stick to it till the end of time, so there's not much motivation to make this too advanced.

Haha, fair enough!

**Comment by MKadaner** [comment]
@ags1234, welcome to Far user community as they say it these days. I highly recommend doing two things:

1. Turn off mouse in `Options` → `Interface settings` → `Mouse`. Far is at its best when controlled from the keyboard.
2. Read help from cover to cover. It is not large. You can even read `FarEng.hlf` file. It's a text file, just about 7000 short lines. The best way to read it is in the built-in Far editor (`F4`). You likely would forget most of the keyboard shortcuts, but you will get an idea of what is possible, so next time you need something, you know it is there somewhere.

**Reply by ags1234** [reply]
I am going to do that, thanks.
There might be a somewhat difficult accommodation to Far for new users, mainly because it is text mode.
So no comfortable interaction with the rest of the GUI software, like drag-and-drop.
Also the window space is not as efficiently used when all the fonts are fixed width.
But the interface is clear, very well organized, with a lot of advanced options. It shows it's maturity.
With a nice theme, there is a certain sweetness to Far.

**Reply by MKadaner** [reply]
> ... because it is text mode.

Well, it's the whole point of using Far. In exchange, you're getting a real, honest command prompt and console. You can run (virtually) any command line application, and it will work exactly as you would expect it. I believe there is no way to emulate console in a GUI application with this level of fidelity (unless you use another console emulator). For example, if you use a command line script to set up a complex command-line-based development environment (which is often the case), you can launch Far at the end of such script and have the fully functional environment plus Far panels and all other goodies.

> ... like drag-and-drop.

While you cannot drop a file from Explorer to Far (but why do you use Explorer to start with? 🙂) you can copy file's full path with `Alt+Shift+Ins` and paste it directly to any GUI "Open" or "Save" dialog. This trick replaces dragging file from Far to GUI applications and works like charm.

**Reply by ags1234** [reply]
You are right about Explorer :)
But unfortunately the Desktop is still Explorer.
I found this plugin (https://github.com/VictorVG/PLG/blob/master/Drag-n-Drop.7z), which should add support for drag-and-drop to Far.
The problem is that when I try to drag a file from Far to Desktop it badly freezes my ConEmu+Far. But if I am patient through the freeze, the control is returned and the file is copied.

**Reply by bitraid** [reply]
If you are using ConEmu, it already provides support for drag and drop with Far. Look at `Settings -> Far Manager`.

**Comment by ags1234** [comment]
@bitraid, thanks for that. It seems there is a tight integration between these two.
I disabled the plugin and relied only on ConEmu for drag-and-drop, but it doe not work for me, in neither direction.

**Reply by bitraid** [reply]
Have you installed the ConEmu plugin in Far?

**Reply by ags1234** [reply]
@bitraid
Hey, it worked!
I copied ConEmu directory from \ConEmu\plugins into \Far\Plugins. And restarted Far.
Now in F11 menu I have 4 new ConEmu entries.

**Comment by ags1234** [comment]
> In theory we could introduce a proper UI for this, with a list of available themes, preview, export etc. I even have a draft branch somewhere.

@alabuzhev Thank you for introducing this very useful UI menu starting build 6444 ;)

For people getting here from search engines, I paste from changelog:

--------------------------------------------------------------------------------
drkns 2025-02-26 20:59:10+00:00 - build 6444

1. Experimental: custom color themes (%FARHOME%\Addons\Colors\Interface) can be selected directly from the UI (Options - Colors - Themes).
--------------------------------------------------------------------------------
